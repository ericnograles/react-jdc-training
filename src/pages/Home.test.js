import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Home from './Home';

describe('<Home>', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    let login = jest.fn();
    let auth = {
      isAuthenticated: false
    };
    let location = {
      state: { from: '/protected' }
    };
    ReactDOM.render(
      <Home login={login} auth={auth} location={location} />,
      div
    );
  });

  it('properly shows login when not authenticated', () => {
    let login = jest.fn();
    let auth = {
      isAuthenticated: false
    };
    let location = {
      state: { from: '/protected' }
    };
    const output = shallow(
      <Home login={login} auth={auth} location={location} />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
