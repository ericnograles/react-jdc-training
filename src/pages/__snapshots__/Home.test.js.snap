// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home> properly shows login when not authenticated 1`] = `
<div>
  <p
    className="App-intro"
  >
    <button
      onClick={[Function]}
    >
      Fake Login
    </button>
  </p>
</div>
`;
